src/neon/deser.rs

```diff
@@ -0,0 +1,204 @@
+use std::mem;
+
+pub use crate::error::{Error, ErrorType};
+pub use crate::Deserializer;
+pub use crate::Result;
+pub use crate::neon::intrinsics::*;
+pub use crate::neon::utf8check::*;
+pub use crate::stringparse::*;
+
+use crate::neon::intrinsics::*;
+
+
+impl<'de> Deserializer<'de> {
+    #[cfg_attr(not(feature = "no-inline"), inline(always))]
+    pub fn parse_str_(&mut self) -> Result<&'de str> {
+        // Add 1 to skip the initial "
+        let idx = self.iidx + 1;
+        let mut padding = [0u8; 32];
+        //let mut read: usize = 0;
+
+        // we include the terminal '"' so we know where to end
+        // This is safe since we check sub's lenght in the range access above and only
+        // create sub sliced form sub to `sub.len()`.
+
+
+        //
+        // FIXME - this section is somewhat questionable -- needs a complete proofread!!!
+        //
+
+        let src: &[u8] = unsafe { &self.input.get_unchecked(idx..) };
+        let mut src_i: usize = 0;
+        let mut len = src_i;
+        loop {
+            let v: int8x16_t = if src.len() >= src_i + 16 {
+                // This is safe since we ensure src is at least 16 wide
+                #[allow(clippy::cast_ptr_alignment)]
+                    unsafe {
+                    *(src.as_ptr().add(src_i) as *const int8x16_t)
```

Comment:
these were supposed to be load instructions

Reply:
`vld1q_s8` should do the trick here?